camus-api/src/test/java/com/linkedin/camus/schemaregistry/CachedSchemaRegistryTest.java

```diff
@@ -0,0 +1,69 @@
+package com.linkedin.camus.schemaregistry;
+
+import java.util.Properties;
+
+import org.apache.avro.Schema;
+import org.junit.Before;
+import org.junit.Test;
+import org.easymock.EasyMock;
+
+
+public class CachedSchemaRegistryTest {
+  private static final String GET_SCHEMA_BY_ID_MAX_RETIRES = "get.schema.by.id.max.retries";
+  private static final String GET_SCHEMA_BY_ID_MIN_INTERVAL_SECONDS = "get.schema.by.id.min.interval.seconds";
+  private CachedSchemaRegistry<Schema> cachedRegistry;
+  private SchemaRegistry<Schema> registry;
+  private Properties props;
+
+  @SuppressWarnings("unchecked")
+  @Before
+  public void setupRegistryMock() {
+    props = new Properties();
+    registry = EasyMock.createNiceMock(SchemaRegistry.class);
+    EasyMock.expect(registry.getSchemaByID(EasyMock.anyString(), EasyMock.anyString())).andThrow(
+        new SchemaNotFoundException());
+  }
+
+  @Test
+  public void testMaxRetries() {
+    EasyMock.expectLastCall().times(20);
```

Comment:
nit: I find it a bit strange to have the expect() method in setupRetistryMock() and then have the .times(20) here. It may create unexpected behavior if someone adds another expect() call in setupRetistryMock() . I think it is better to keep the expect() call close to times() even if it means to duplicate one line of code.